tests/testthat/test-across.R

```diff
@@ -888,6 +888,32 @@ test_that("if_any() and if_all() expansions deal with no inputs or single inputs
   )
 })
 
+test_that("if_any() on zero-column selection behaves as expected in filter", {
+  tbl <- tibble(
+    x1 = 1:5,
+    x2 = c(-1, 4, 5, 4, 1),
+    y = c(1, 4, 2, 4, 9),
+  )
+
+  expect_equal(
+    filter(tbl, if_any(c(), ~ is.na(.x))),
+    filter(tbl, FALSE)
```

Comment:
Expected values seem slightly better without too much logic, not sure if it's easier to read though:

```suggestion
    tbl[0, ]
```

Reply:
Thanks for the feedback!